libs/libdaemon/src/BaseDaemon.cpp

```diff
@@ -603,6 +603,9 @@ void BaseDaemon::reloadConfiguration()
       */
     config_path = config().getString("config-file", "config.xml");
     loaded_config = ConfigProcessor(config_path, false, true).loadConfig(/* allow_zk_includes = */ true);
+
+    ConfigProcessor::main_config_path = Poco::Path(config_path).makeParent().toString();
```

Comment:
Using global variables with non-trivial constructor across translation units is undefined behaviour.